iOSDFULibrary/Classes/Utilities/DFUPackage/Manifest/SoftdeviceBootloaderInfo.swift

```diff
@@ -30,17 +30,47 @@
 
 import Foundation
 
-class SoftdeviceBootloaderInfo: ManifestFirmwareInfo {
-    var blSize: UInt32 = 0
-    var sdSize: UInt32 = 0
+// MARK: - SoftdeviceBootloaderInfo
+
+struct SoftdeviceBootloaderInfo: ManifestFirmware, Codable {
     
-    override init(withDictionary aDictionary : Dictionary<String, AnyObject>) {
-        super.init(withDictionary: aDictionary)
-        if aDictionary.keys.contains("bl_size") {
-            blSize = (aDictionary["bl_size"]!).uint32Value
-        }
-        if aDictionary.keys.contains("sd_size") {
-            sdSize = (aDictionary["sd_size"]!).uint32Value
+    // MARK: Properties
+    
+    var binFile: String?
+    var datFile: String?
+    var metadata: Metadata
+    
+    var blSize: UInt32 {
+        metadata.blSize
+    }
+    
+    var sdSize: UInt32 {
+        metadata.sdSize
+    }
+    
+    // MARK: CodingKeys
+    
+    enum CodingKeys: String, CodingKey {
+        case binFile = "bin_file"
+        case datFile = "dat_file"
+        case metadata = "info_read_only_metadata"
```

Comment:
`metadata` is optional and only for Secure DFU. We actually don't need the sizes for Secure DFU at all, as they are included in the signed `.dat` file.